src/librustc/middle/liveness.rs

```diff
@@ -1398,7 +1399,11 @@ fn check_local<'a, 'tcx>(this: &mut Liveness<'a, 'tcx>, local: &'tcx hir::Local)
         },
         None => {
             this.pat_bindings(&local.pat, |this, ln, var, sp, id| {
-                this.warn_about_unused(sp, id, ln, var);
```

Comment:
Instead of the if let you could prepend this line with `let sp = local.pat.simple_span.unwrap_or(sp);`